ui/app/store/actions.js

```diff
@@ -1191,21 +1191,24 @@ export function setSelectedAddress (address) {
 }
 
 export function showAccountDetail (address) {
-  return (dispatch) => {
+  return async (dispatch) => {
     dispatch(showLoadingIndication())
     log.debug(`background.setSelectedAddress`)
-    background.setSelectedAddress(address, (err, tokens) => {
+
+    let tokens
+    try {
+      tokens = await promisifiedBackground.setSelectedAddress(address)
+    } catch (error) {
       dispatch(hideLoadingIndication())
-      if (err) {
-        return dispatch(displayWarning(err.message))
-      }
-      dispatch(updateTokens(tokens))
-      dispatch({
-        type: actionConstants.SHOW_ACCOUNT_DETAIL,
-        value: address,
-      })
-      dispatch(setSelectedToken())
+      return dispatch(displayWarning(error.message))
```

Comment:
Should we separate this into two lines—I believe the old return value was `undefined`, even when there was an error

Reply:
Good catch! I meant to check for this since you pointed this out in the other PR.... I forgot with these two.

Reply:
This has been updated